code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -739,6 +739,29 @@ emp_act
 /mob/living/carbon/human/water_act(volume, temperature, source, method = REAGENT_TOUCH)
 	. = ..()
 	dna.species.water_act(src, volume, temperature, source, method)
+	if(method != REAGENT_TOUCH)
+		return
+
+	head?.water_act(src, volume, temperature, source, method)
+	wear_suit?.water_act(src, volume, temperature, source, method)
+	l_hand?.water_act(src, volume, temperature, source, method)
+	r_hand?.water_act(src, volume, temperature, source, method)
+	if((head?.flags & THICKMATERIAL) && (wear_suit?.flags & THICKMATERIAL)) // fully pierce proof clothing is also water proof!
+		return
+	else
+		w_uniform?.water_act(src, volume, temperature, source, method)
```

Comment:
I think you might be better served by putting these all into a list and doing this iteratively

Reply:
Done!